lxd/network/driver_ovn.go

```diff
@@ -5442,8 +5442,30 @@ func (n *ovn) Leases(projectName string, clientType request.ClientType) ([]api.N
 	var err error
 	leases := []api.NetworkLease{}
 
-	// If requested project matches network's project then include gateway IPs.
-	if projectName == n.project || projectName == "" {
+	var project *api.Project
+
+	// Only query for project when we need to check for "features.networks". That is, when the network project is
+	// the default project and projectName refers to any one other project.
+	if projectName != api.ProjectDefaultName && projectName != "" && n.project == api.ProjectDefaultName {
```

Comment:
Can we move this logic into `NetworkProject` to avoid the duplication? Would that be correct?

Reply:
I don't think this would be correct since we have to consider the networks from project `default` and the instances from the requested project (e.g. `projectName`)

Reply:
Lets move this to a function in network.Common